src/magellan_vision/scripts/line_detector.py

```diff
@@ -1,211 +1,170 @@
+            self._MAXLINEGAP = rospy.get_param("max_line_gap")
+        except KeyError as e:
+            rospy.logerr("LineDetector: error in looking up param. {}".format(e))
+            raise
+
+    def skeletize(self, img, size, skel):
+        element = cv2.getStructuringElement(cv2.MORPH_CROSS, (3, 3))
+        while not rospy.is_shutdown():
+            eroded = cv2.erode(img, element)
+            temp = cv2.dilate(eroded, element)
+            temp = cv2.subtract(img, temp)
+            skel = cv2.bitwise_or(skel, temp)
+            img = eroded.copy()
+            zeros = size - cv2.countNonZero(img)
+            if zeros == size:
+                break
+
+        return skel
+
+    def detect(self, cv_image):
+        gray = cv2.cvtColor(cv_image, cv2.COLOR_BGR2GRAY)
+        blurred = cv2.GaussianBlur(gray, (3, 3), 0)
+        ret, thresh4 = cv2.threshold(blurred, 200, 255, cv2.THRESH_TOZERO)
+        size = np.size(thresh4)
+        skel = np.zeros(thresh4.shape, np.uint8)
+
+        skeleton = self.skeletize(thresh4, size, skel)
+
+        dst = cv2.Laplacian(skeleton, self._ddepth, ksize=self._KERNEL_SIZE)
+        abs_dst = cv2.convertScaleAbs(dst)
+
+        linesP = cv2.HoughLinesP(abs_dst, rho=self._RHO, theta=self._THETA, threshold=self._THRESHOLD,
+                                 lines=np.array([]), minLineLength=self._MINLINELENGTH, maxLineGap=self._MAXLINEGAP)
+
+        if linesP is not None:
+            for line in linesP:
+                for x1, y1, x2, y2 in line:
+                    slope = (y2 - y1) / (x2 - x1)
+                    p1 = Point()
+                    p2 = Point()
+                    p1.x = x1
+                    p1.y = y1
+                    p1.z = 0
+                    point_arr.extend(p1)
+                    p2.x = x2
+                    p2.y = y2
+                    p2.z = 0
+                    point_arr.extend(p2)
+                    # <-- Calculating the slope.
+                    if math.fabs(slope) < .5:
+                        # <-- Only consider extreme slope
+                        continue
+                    if x1 < cv_image.shape[1]/2 and x2 < cv_image.shape[1]/2:
+                        # <-- If the slope is negative, left group
+                        cv2.circle(cv_image, (x1, y1), (5), (0, 0, 255), 3)
+                        cv2.circle(cv_image, (x2, y2), (5), (0, 0, 255), 3)
+                    else:  # <-- Otherwise, right group.
+                        cv2.circle(cv_image, (x1, y1), (5), (0, 255, 0), 3)
+                        cv2.circle(cv_image, (x2, y2), (5), (0, 255, 0), 3)
+        return cv_image
+
+
+def main(args):
+    rospy.init_node('LinesNode')
+    node_ = PubSubNode()
+    r = rospy.get_param("rate")
+    rate = rospy.Rate(r)
+    try:
+        while not rospy.is_shutdown():
+            node_.run_detects()
+            rate.sleep()
+
+    except:
+        rospy.logfatal("shutting down ROS Lines detector Module")
```

Comment:
Would be nice to print the exception, since this is basically a catch-all for the entire program